pandas/tests/frame/test_analytics.py

```diff
@@ -466,6 +466,58 @@ def test_corrwith_mixed_dtypes(self):
         expected = pd.Series(data=corrs, index=['a', 'b'])
         tm.assert_series_equal(result, expected)
 
+    def test_corrwith_multindex(self):
+        # PR #22622
+        df = pd.DataFrame(np.random.random((4,4)))
+        df.columns = pd.MultiIndex.from_product([[1,2],['A','B']])
+
+        # must raise an exception if columns use MultiIndex
+        try:
+            df.corrwith(df)
```

Comment:
these try/except's don't make any sense, use ``with pytest.raises`` if you are trying to assert that it raises